libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1446,6 +1446,28 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 		_functionCall.annotation().type = make_shared<TupleType>(functionType->returnParameterTypes());
 
 	TypePointers parameterTypes = functionType->parameterTypes();
+
+	if (!_functionType.padArguments())
+	{
+		for (size_t i = 0; i < arguments.size(); ++i)
```

Comment:
Could use range-based loop, but is also fine this way.

Reply:
This file makes heavy use of this, probably should just make one giant cleanup PR changing all?